libr/flag/flag.c

```diff
@@ -725,15 +778,52 @@ R_API void r_flag_bind(RFlag *f, RFlagBind *fb) {
 	fb->pop_fs = r_flag_space_pop;
 }
 
+static bool flag_count_foreach(RFlagItem *fi, void *user) {
+	int *count = (int *)user;
+	(*count)++;
+	return true;
+}
+
 R_API int r_flag_count(RFlag *f, const char *glob) {
 	int count = 0;
-	RFlagItem *flag;
-	RListIter *iter;
 	r_return_val_if_fail (f, -1);
-	r_list_foreach (f->flags, iter, flag) {
-		if (r_str_glob (flag->name, glob)) {
-			count++;
-		}
-	}
+
+	r_flag_foreach_glob (f, glob, flag_count_foreach, &count);
 	return count;
 }
+
+#define FOREACH_BODY(condition) \
```

Comment:
what about making a small static function implementing the condition and make a function instead of a multiline macro here?

Reply:
Mmm... I'm not sure it's possible. The condition needs to access some parameters that are different for each function, so I would need to do things in a weird way. The macro makes it easier. Do you have a solution for this?

Reply:
Please share how you would do that. I cannot think of a way to do that without complicating things for no reasons.